File: InitializrSpringbootProject/src/main/java/com/seminario/integrador/pawplan/controller/values/ReservarTurnoRq.java
package com.seminario.integrador.pawplan.controller.values;

import java.sql.Date;

import lombok.Data;

@Data
public class ReservarTurnoRq {
    private Long veterinariaId;
    private Long veterinarioId;
    private boolean esDomicilio;
    private long animalId;
    private String fecha;
    private String hora;

    //================

    private String dniCliente; //En caso de necesitar crear un nuevo cliente
    private AnimalRq animalRq; //En caso de necesitar crear un nuevo animal
}

File: PawPanFront/src/app/model/ReservarTurnoRq.ts
import { AnimalRq } from "./AnimalRq";

export class ReservarTurnoRq {
    veterinariaId: number;
    veterinarioId: number;
    animalId: number;
    fecha: string;
    hora: string;
    esDomicilio: boolean;

    dniCliente:string;  //En caso de necesitar crear un nuevo cliente
    animalRq: AnimalRq; //En caso de necesitar crear un nuevo animal
}